### ISSUE TITLE
NullReferenceException in 1.1.7

### ISSUE THREAD

**Issue by bschwehn** [lateapexearlyspeed/Lateapexearlyspeed.JsonSchema#43]
While the validation works fine in my unittests, I get a NRE when trying to create a JsonValidator from an blazor application.

From debuggging I can see that this returns null: https://github.com/lateapexearlyspeed/Lateapexearlyspeed.JsonSchema/blob/master/LateApexEarlySpeed.Json.Schema/Common/JsonSerializerOptionsExtensions.cs#L16-L16

and the NRE then comes here:
https://github.com/lateapexearlyspeed/Lateapexearlyspeed.JsonSchema/blob/master/LateApexEarlySpeed.Json.Schema/Keywords/JsonConverters/PropertiesKeywordJsonConverter.cs#L35-L35

But I am not sure yet why this behaviour is only triggered from the web application.

This error does not happen in 1.1.6 which did not have the Validator options.

**Comment by bschwehn** [comment]
I wanted to create a small reprocase, but having a hard time doing so. Maybe this already helps:

When GetJsonValidatorOptions is called, the ConditionalWeakTable JsonValidatorOptionTable contains one entry with key of type JsonSerializerOptions, but it is not the same as parameter "options", so it is not returned with TryGetValue

![image](https://github.com/user-attachments/assets/617941ec-d7ca-4c73-b8b6-a32a22a66c5f)

I don't really understand the purpose of using the CWT here however.

**Comment by bschwehn** [comment]
Ah, I see what is causing it now: The web application referenced Microsoft.Extensions.Logging v8.0.0 (which in turn then causes System.Text.Json v8.0.0 to be used).

Downgrading that package to v6.0.0 makes the issue go away.

Maybe the code could be changed to make it more compatible with v8, but for me this solves the issue for now.

Thanks

**Comment by lateapexearlyspeed** [comment]
Hi @bschwehn thanks for detailed investigation.

> I don't really understand the purpose of using the CWT here

Current implementation needs way to provide JsonValidatorOptions to custom JsonConverters, the JsonSerializerOptions is natural stuff to provide this information into JsonConverter, but System.Text.Json dependency version of this library doesn't support state for deserialization. So I used CWT to associate JsonValidatorOptions instance with JsonSerializerOptions instance.


> Downgrading that package to v6.0.0 makes the issue go away.

So based on your investigation, after using System.Text.Json v8, the JsonSerializerOptions instance passed to JsonConverter is not same as the one passed to deserialization callsite: `JsonSerializer.Deserialize(json, options)`, right ?

**Comment by bschwehn** [comment]
Hi, thanks for the explanation!

>  JsonSerializerOptions instance passed to JsonConverter is not same as the one passed to deserialization callsite

Right, I've added a demo "BlazorApp2" here: https://github.com/bschwehn/Lateapexearlyspeed.JsonSchema/tree/bug/43-jsonserializeroptions

It does not happen in the "ConsoleTest" I added there also, so it is not *just* adding a reference to a v8.0.0 package that is required to triggering it.

I don't think adding v8 packages to a .net 6 application is a valid use case, and it works fine if I upgrade to .net 8 also. So I think the bug is probably invalid.

Thanks for the great library, it works well in what I tested so far!
